x-pack/plugins/cases/public/methods/use_cases_toast.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonEmpty, EuiText } from '@elastic/eui';
+import React from 'react';
+import styled from 'styled-components';
+import { toMountPoint } from '../../../../../src/plugins/kibana_react/public';
+import { Case } from '../../common';
+import { useKibana } from '../common/lib/kibana';
+import { CasesDeepLinkId, generateCaseViewPath } from '../common/navigation';
+import { useCasesContext } from '../components/cases_context/use_cases_context';
+import { CASE_SUCCESS_SYNC_TEXT, CASE_SUCCESS_TOAST, VIEW_CASE } from './translations';
+
+const LINE_CLAMP = 3;
+const Title = styled.span`
+  text-overflow: ellipsis;
+  display: -webkit-box;
+  -webkit-line-clamp: ${LINE_CLAMP};
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+`;
+const EuiTextStyled = styled(EuiText)`
+  ${({ theme }) => `
+    margin-bottom: ${theme.eui?.paddingSizes?.s ?? 8}px;
+  `}
+`;
+
+export const useCasesToast = () => {
+  const { appId } = useCasesContext();
+  const {
+    notifications: { toasts },
+    application: { navigateToApp },
+  } = useKibana().services;
+
+  return {
+    showSuccessAttach: (theCase: Case) => {
+      const onViewCaseClick = () => {
+        navigateToApp(appId, {
```

Comment:
You can use the `useCaseViewNavigation` from `x-pack/plugins/cases/public/common/navigation/hooks.ts`.

```
const { getCreateCaseUrl, navigateToCreateCase } = useCaseViewNavigation();

showSuccessAttach: (theCase: Case) => {
      const onViewCaseClick = () => {
          navigateToCaseView({ detailName: theCase.id });
      }
}
```

Reply:
will that take in consideration the appId already? if so, I will move to that.

Reply:
Yes, it will automatically take care of the application ID and everything.

Reply:
These hooks don't offer the same functionality I need for this specific case. I'll leave it as it is.